src/record/record-transition.js

```diff
@@ -67,6 +68,31 @@ RecordTransition.prototype.hasVersion = function( version ) {
 };
 
 /**
+ * Send version exists error if the record has been already loaded, else
+ * store the version exists error to send to the sockerWrapper once the 
+ * record is loaded
+ *
+ * @param   {SocketWrapper} socketWrapper the sender
+ * @param   {Number} version The version number
+ *
+ * @public
+ */
+RecordTransition.prototype.sendVersionExists = function( socketWrapper, version ) {
```

Comment:
This was based on the previous logic, which is that the last version in the steps ( updates added  ) is assumed to be the records current version. I'm guessing this was to assume a positive path ( which is usually correct ).